### PR TITLE
Add vincepri to Cluster API GCP repo admin / update maintainers

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

### CODE REVIEW

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml [hunk 1]
```diff
@@ -113,11 +113,16 @@ teams:
...
     - roberthbailey
     - timothysc
+    - vincepri
     privacy: closed
   cluster-api-provider-gcp-maintainers:
     description: write access to cluster-api-provider-gcp-maintainers
     members:
+    - detiber
     - justinsb
+    - krousey
```
Comment (timothysc) [inline config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml]: I haven't seen kris around in O(year+)

Comment (vincepri) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml]: I can update if necessary, this is copied from the repo's maintainer list

**Comment by justinsb** [comment]
I'm great with adding @vincepri - thanks & welcome!

On the removals, I'll ask the people but I suggest we should do them separately...

/approve
/lgtm
/hold cancel

**Pull request merged**